cmd/registry/main.go

```diff
@@ -17,8 +18,34 @@ import (
 	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
 	_ "github.com/docker/distribution/registry/storage/driver/s3-goamz"
 	_ "github.com/docker/distribution/registry/storage/driver/swift"
+	"github.com/spf13/cobra"
 )
 
+var subCommands = map[string]*cobra.Command{
+	registry.GCCmd.Name(): registry.GCCmd,
+}
+
+func findSubcommand(subCommands map[string]*cobra.Command, args []string) *cobra.Command {
+	if len(args) < 2 {
+		return nil
+	}
+
+	potentialSubCommand := args[1]
+	if command, ok := subCommands[potentialSubCommand]; ok {
+		return command
+	}
+
+	return nil
+}
+
 func main() {
+	// check if first arg is a subcommand. If not, assume args are for the main
+	// command that starts up a registry server
+	subcommand := findSubcommand(subCommands, os.Args)
```

Comment:
Repeating this since github is hiding this comment for some reason.

This might break global arguments.

At this point, I think it might be better if we break compatibility, rather than adopt a flaky approach:

```go
	cmd := cobra.Command{
		Use: filepath.Base(os.Args[0]),
	}

	servecmd := &cobra.Command{
		Use: "serve",
		Run: func(cmd *cobra.Command, args []string) {
			fmt.Println("serve", args)
		},
	}

	gccmd := &cobra.Command{
		Use: "gc",
		Run: func(cmd *cobra.Command, args []string) {
			fmt.Println("gc", args)
		},
	}

	cmd.AddCommand(servecmd, gccmd)

	if err := cmd.Execute(); err != nil {
		log.Fatalf("%v", err)
	}
```

Then, we can just have a small bash script that rewrites to `registry <config.yml>` to `registry serve <config.yml>`. Or, better yet, the entrypoint can simply be `registry serve` by default and no one will be the wiser.

Sorry to hang this PR up on this point. I'd like to add other management commands in the future, so getting this right is important.

@RichardScothern What do you think?